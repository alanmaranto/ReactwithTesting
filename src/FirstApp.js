import React from "react";
import PropTypes from "prop-types";

const FirstApp = ({ greeting, subtitle }) => {
  return (
    <>
      <h1>{greeting}</h1>
      <p>{subtitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  greeting: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

FirstApp.defaultProps = {
  subtitle: "Subtitle",
};

export default FirstApp;
