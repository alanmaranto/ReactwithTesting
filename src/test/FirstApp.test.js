/* import React from "react";
import { render } from "@testing-library/react";
import FirstApp from "../FirstApp";

describe("Tests in FirstApp Component", () => {
  test('should display the message "hello', () => {
    const greeting = "hello";
    const { getByText } = render(<FirstApp greeting={greeting} />);

    expect(getByText(greeting)).toBeInTheDocument();
  });
});
 */

import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import FirstApp from "../FirstApp";

describe("Tests in First A", () => {
  test("should display First App correctly", () => {
    const greeting = "hello";
    const wrapper = shallow(<FirstApp greeting={greeting} />);

    expect(wrapper).toMatchSnapshot()
  });
});
