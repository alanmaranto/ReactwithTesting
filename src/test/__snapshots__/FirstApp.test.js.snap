// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in First A should display First App correctly 1`] = `
<Fragment>
  <h1>
    hello
  </h1>
  <p>
    Subtitle
  </p>
</Fragment>
`;
